### ISSUE TITLE
new autodoc warnings on 1.7.0b2

### ISSUE THREAD

**Issue by abchenson** [sphinx-doc/sphinx#4573]
Subject: new autodoc warnings on 1.7.0b2
### Problem
- on switching from 1.6.6 to 1.7.0b2, modules which were previously being documented properyl by autosummary are now throwing warnings and failing to compile properly:
e.g. for a method mock_code.mock_mod1.aaa.aaa_a in class mock_code.mock_mod1.aaa, 
```
WARNING: autodoc: failed to import data '__init__' from module 'mock_code.mock_mod1.aaa'; the following exception was raised:
No module named 'mock_code.mock_mod1.aaa'; 'mock_code.mock_mod1' is not a package
```
The command being used is
```
.. currentmodule:: mock_code.mock_mod1

.. autosummary::
    :toctree: generated/

    aaa
```
.  This works fine on 1.6.6, generating and linking the files properly.  With 1.7.0b2 no link from the class page appears for the method.

#### Procedure to reproduce the problem
```
make html
```

If you find it difficult to reproduce this I can try to make a minimal example (actually this started to happen while I was trying to make an example for something else!)

**Comment by tk0miya** [comment]
Thank you for reporting.
It would be nice if you can provide a minimal reproducible example.

Thanks,

**Comment by abchenson** [comment]
[example.tar.gz](https://github.com/sphinx-doc/sphinx/files/1704147/example.tar.gz)

This minimal example is based on running sphinx-quicktstart, adding what I need to conf.py for autosummary to work, and some breif mocked up code. Switching between 1.6.6. and 1.7.0b and running make html switches a warning on and off, the same warning I was seeing lots of examples of in a big project (when running I deleted the build and generated directories first).  I had to include some autodocs templates for the problem to reoccurr (they are not very much different from the default so this is strange).
